### PR TITLE
Add The Dawn of Delectability quest

### PR DESCRIPTION
**Pull request by Era-Lusiphur**
Adds the Crystal War quest The Dawn of Delectability, which rewards the Trainee Knife.

https://www.bg-wiki.com/ffxi/The_Dawn_of_Delectability

Requires items global additions from #559

**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

### CODE REVIEW

File: scripts/quests/crystalWar/The_Dawn_of_Delectability.lua [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 2
+        end,
+
+        [xi.zone.WINDURST_WATERS] =
+        {
+            ['Ranpi-Monpi'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHas(trade, {xi.items.PEPPERONI, xi.items.WALNUT,
```
Comment (claywar) [inline scripts/quests/crystalWar/The_Dawn_of_Delectability.lua]: Probably should be `tradeHasExactly`

File: scripts/quests/crystalWar/The_Dawn_of_Delectability.lua [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+                    return quest:progressEvent(979) -- reminder dialogue
+                end,
+            },
+
+            onEventFinish =
+            {
+                [983] = function(player, csid, option, npc)
+                    player:confirmTrade()
+                    quest:setVar(player, 'Prog', 3)
+                    quest:setVar(player, "DelectabilityDay", vanaDay())
```
Comment (claywar) [inline scripts/quests/crystalWar/The_Dawn_of_Delectability.lua]: If you're looking for a unique day, there's a binding for this in luautils as well, `VanadielUniqueDay`

File: scripts/quests/crystalWar/The_Dawn_of_Delectability.lua [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+        [xi.zone.WINDURST_WATERS] =
+        {
+            ['Ranpi-Monpi'] = quest:progressEvent(980),
+
+            onEventFinish =
+            {
+                [980] = function(player, csid, option, npc)
+                    npcUtil.giveKeyItem(player, xi.ki.RANPIMONPI_SPECIALTY)
+                    quest:messageSpecial(watersID.text.KEYITEM_OBTAINED, xi.ki.RANPIMONPI_SPECIALTY)
```
Comment (claywar) [inline scripts/quests/crystalWar/The_Dawn_of_Delectability.lua]: Shouldn't need this messageSpecial using giveKeyItem

File: scripts/quests/crystalWar/The_Dawn_of_Delectability.lua [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+        [xi.zone.WINDURST_WATERS_S] =
+        {
+            ['Ranpi-Monpi'] = quest:progressEvent(118),
+
+            onEventFinish =
+            {
+                [118] = function(player, csid, option, npc)
+                    player:delKeyItem(xi.ki.RANPIMONPI_SPECIALTY)
+                    npcUtil.giveKeyItem(player, xi.ki.CULINARY_KNIFE)
+                    quest:messageSpecial(watersID.text.KEYITEM_OBTAINED, xi.ki.CULINARY_KNIFE)
```
Comment (claywar) [inline scripts/quests/crystalWar/The_Dawn_of_Delectability.lua]: Similar, shouldn't be necessary

File: scripts/quests/crystalWar/The_Dawn_of_Delectability.lua [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+
+        [xi.zone.WINDURST_WATERS] =
+        {
+            ['Ranpi-Monpi'] = quest:progressEvent(981),
+
+            onEventFinish =
+            {
+                [981] = function(player, csid, option, npc)
+                    player:delKeyItem(xi.ki.CULINARY_KNIFE)
+                    quest:complete(player)
```
Comment (claywar) [inline scripts/quests/crystalWar/The_Dawn_of_Delectability.lua]: There should probably be some safety here.  We don't want to delete their keyitem unless the quest successfully completes (it won't if the player's inventory is full)
```
if quest:complete(player) then
    player:delKeyItem(xi.ki.CULINARY_KNIFE)
end
```

**Comment by Era-Lusiphur** [comment]
Made requested changes, tested.

File: scripts/quests/crystalWar/The_Dawn_of_Delectability.lua [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+                end,
+            },
+        },
+    },
+
+    -- Speak with Ranpi before day is up
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 3 and
+                quest:getVar(player, "Timer") > vanaDay()
```
Comment (claywar) [inline scripts/quests/crystalWar/The_Dawn_of_Delectability.lua]: Similar to other PR, consistency for VanadielUniqueDay with comparison

File: scripts/quests/crystalWar/The_Dawn_of_Delectability.lua [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+        {
+            ['Ranpi-Monpi'] = quest:progressEvent(982)
+        },
+    },
+
+    -- Speak with Ranpi after a game day wait
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 3 and
+                quest:getVar(player, "Timer") <= vanaDay()
```
Comment (claywar) [inline scripts/quests/crystalWar/The_Dawn_of_Delectability.lua]: And here

**Review by claywar — APPROVED** [review]

**Pull request merged**